src/dma.c

```diff
@@ -47,6 +47,54 @@ static volatile int __dma_busy(void)
     return PI_regs->status & (PI_STATUS_DMA_BUSY | PI_STATUS_IO_BUSY);
 }
 
+/**
+ * @brief Check whether the specified PI address can be accessed doing I/O from CPU
+ * 
+ * The PI bus covers the full 32-bit address range. The full range is only accessible
+ * via DMA, though. A part of the range is also memory mapped to the CPU and can be
+ * accessed via #io_read and #io_write.
+ * 
+ * The ranges of PI address that can be access via CPU are:
+ * 
+ *  * 0x0500_0000 - 0x0FFF_FFFF: used by N64DD and SRAM on cartridge
+ *  * 0x1000_0000 - 0x1FBF_FFFF: cartridge ROM
+ *  * 0x1FD0_0000 - 0x1FFF_FFFF: no known PI peripherals use this
+ * 
+ * The rest of the 32-bit address range is only accessible via DMA.
+ * 
+ * Notice also that the range 0x2000_0000 - 0x7FFF_FFFF is theoretically accessible
```

Comment:
Just a question: why is it only accessible via 64 bit addressing?

Reply:
In 32-bit mode, addresses are made of 29-bit address + 3 bit of segment. The direct-mapped segments like what we call CachedAddr / UncachedAddr only allow access to physical address up to 29 bits, that is 0x1FFF_FFFF. To access addresses in the full 32-bit physical range you either need TLBs, or 64-bit addressing.